### PR TITLE
Update reward module to emit an event - Closes #7287

### LINKED ISSUE #7287: Update reward module to emit an event
### Description

Based on the [PR](https://github.com/LiskHQ/lips/pull/147), update reward module.

* Add event type constants.
  * `TYPE_ID_REWARD_MINTED: Buffer.from([0,0,0,1])`
  * `REWARD_NO_REDUCTION: 0`
  * `REWARD_REDUCTION_SEED_REVEAL: 1`
  * `REWARD_REDUCTION_MAX_PREVOTES: 2`
* Add rewardMinted event
Topic
```
generatorAddress: the address of the block generator that obtains the reward.
```
Data
```
rewardMintedDataSchema = {
    "type": "object",
    "required": ["amount", "reduction"],
    "properties": {
        "amount": {
            "dataType": "uint64",
            "fieldNumber": 1
        },
        "reduction": {
            "dataType": "uint32",
            "fieldNumber": 2
        }
    }
}
```
amount: the amount of rewards minted.
reduction: an integer indicating whether the reward was reduced and for which reason. Allowed values are: `REWARD_NO_REDUCTION`, `REWARD_REDUCTION_SEED_REVEAL`, `REWARD_REDUCTION_MAX_PREVOTES`.

* Update `getBlockReward` API to return event type
```python
def getBlockReward(blockHeader: BlockHeader) -> tuple[uint64, uint32]:
  if isSeedRevealValid(blockHeader.generatorAddress, blockHeader.seedReveal) == False:
    return (0, REWARD_REDUCTION_SEED_REVEAL)
  defaultReward = getDefaultRewardAtHeight(blockHeader.height)
  if impliesMaximalPrevotes(blockHeader) == False:
    return (defaultReward / REWARD_REDUCTION_FACTOR_BFT, REWARD_REDUCTION_MAX_PREVOTES)
  return (defaultReward, REWARD_NO_REDUCTION)
```

* Update `afterBlockExecute` hook to emit event
```
def afterBlockExecute(b: Block):
    (blockReward, reduction) = getBlockReward(b.header)
    if blockReward > 0:
        Token.mint(b.header.generatorAddress, TOKEN_ID_REWARD, blockReward)
    emitEvent(
        moduleID = MODULE_ID_REWARD,
        typeID = TYPE_ID_REWARD_MINTED,
        data = {
            "amount": blockReward,
            "reduction": reduction
            },
        topics = [b.header.generatorAddress]
    )
```

### Acceptance Criteria

* Add tests to check the event is emitted for all 3 cases

### Additional Information
- LIP PR: https://github.com/LiskHQ/lips/pull/147
- [LIP:0042](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0042.md)

### PR DESCRIPTION
**Pull request by sitetester**
### What was the problem?

This PR resolves #7287

### How was it solved?

Reward module is updated to emit `rewardMinted` event with given constants.

### How was it tested?
test cases added in framework/test/unit/modules/reward/reward_module.spec.ts

### CODE REVIEW

File: framework/src/modules/reward/module.ts [hunk 1]
```diff
@@ -103,5 +105,21 @@ export class RewardModule extends BaseModule {
...
+		const topics: Buffer[] = [];
+		topics.push(context.header.generatorAddress);
+
+		const data = codec.encode(rewardMintedDataSchema, rewardMintedData);
+		context.eventQueue.add(
+			this.id,
+			TYPE_ID_REWARD_MINTED,
+			codec.encode(rewardMintedDataSchema, data),
+			topics,
+		);
```
Comment (ishantiw) [inline framework/src/modules/reward/module.ts:123]:
```suggestion
		const data = codec.encode(rewardMintedDataSchema, rewardMintedData);
		context.eventQueue.add(
			this.id,
			TYPE_ID_REWARD_MINTED,
			codec.encode(rewardMintedDataSchema, data),
			[context.header.generatorAddress],
		);
```

File: framework/test/unit/modules/reward/reward_module.spec.ts [hunk 2]
```diff
@@ -83,24 +90,45 @@ describe('RewardModule', () => {
...
+			expect(blockAfterExecuteContext.eventQueue.getEvents()[0].toObject().typeID).toBe(
+				TYPE_ID_REWARD_MINTED,
+			);
+		});
+
+		it('should emit rewardMinted event for event type REWARD_REDUCTION_MAX_PREVOTES', async () => {
+			rewardModule.api.getBlockReward = jest
+				.fn()
+				.mockReturnValue([
+					BigInt(1) / BigInt(REWARD_REDUCTION_FACTOR_BFT),
```
Comment (ishantiw) [inline framework/test/unit/modules/reward/reward_module.spec.ts:125]: why this?
```ts
					BigInt(1) / BigInt(REWARD_REDUCTION_FACTOR_BFT),
```

File: framework/test/unit/modules/reward/reward_module.spec.ts [hunk 3]
```diff
@@ -83,24 +90,45 @@ describe('RewardModule', () => {
...
+			rewardModule.api.getBlockReward = jest
+				.fn()
+				.mockReturnValue([BigInt(0), REWARD_REDUCTION_SEED_REVEAL]);
+			await rewardModule.afterTransactionsExecute(blockAfterExecuteContext);
+			expect(blockAfterExecuteContext.eventQueue.getEvents()[0].toObject().typeID).toBe(
+				TYPE_ID_REWARD_MINTED,
+			);
+		});
+
+		it('should emit rewardMinted event for event type REWARD_REDUCTION_MAX_PREVOTES', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/reward/reward_module.spec.ts:121]: I think case for `REWARD_REDUCTION_FACTOR_BFT` is missing

**Review by ishantiw — CHANGES_REQUESTED** [review]

**Review by ishantiw — APPROVED** [review]
Great job @sitetester 👏

**Review by mitsuaki-u — APPROVED** [review]
minor comment

**Pull request merged**